main.go

```diff
@@ -969,3 +1035,93 @@ func SolutionActionRevert(solName string) {
 	}
 	fmt.Println("Parameters tuned by the notes referred by the SAP solution have been successfully reverted.")
 }
+
+// getFileName returns the corresponding filename of a given noteID
+// additional it returns a boolean value which is pointing out that the Note
+// the Note is a custom Note (extraNote = true) or an internal one
+func getFileName(noteID string) (string, bool) {
+	extraNote := false
+	fileName := fmt.Sprintf("%s%s", NoteTuningSheets, noteID)
+	if _, err := os.Stat(fileName); os.IsNotExist(err) {
+		// Note is NOT an internal Note, but may be a custom Note
+		extraNote = true
+		_, files := system.ListDir(ExtraTuningSheets, "")
+		for _, f := range files {
+			if strings.HasPrefix(f, noteID) {
+				fileName = fmt.Sprintf("%s%s", ExtraTuningSheets, f)
+			}
+		}
+		if _, err := os.Stat(fileName); os.IsNotExist(err) {
+			errorExit("Note %s not found in %s or %s.", noteID, NoteTuningSheets, ExtraTuningSheets)
+		} else if err != nil {
+			errorExit("Failed to read file '%s' - %v", fileName, err)
+		}
+	} else if err != nil {
+		errorExit("Failed to read file '%s' - %v", fileName, err)
+	}
+	return fileName, extraNote
+}
+
+// getovFile returns the corresponding override filename of a given noteID
+// additional it returns a boolean value which is pointing out if the
+// override file already exists (overrideNote = true) or not
+func getovFile(noteID string) (string, bool) {
+	overrideNote := true
+	ovFileName := fmt.Sprintf("%s%s", OverrideTuningSheets, noteID)
+	if _, err := os.Stat(ovFileName); os.IsNotExist(err) {
+		overrideNote = false
+	} else if err != nil {
+		errorExit("Failed to read file '%s' - %v", ovFileName, err)
+	}
+	return ovFileName, overrideNote
+}
+
+// readYesNo asks the user for yes/no answer.
+// "y", "Y", "yes", "YES", and "Yes" following by "enter" count as confirmation
+// "n", "N", "no", "NO", and "No" following by "enter" count as non-confirmation
+func readYesNo(s string, in io.Reader) bool {
```

Comment:
@angelabriel Maybe if the user includes an invalid answer, that is not Yes or No, we should loop and ask again until a valid answer?

Reply:
yes, it is looping. It is an endless 'for' loop, which is ended by the 'return' statement, which returns true for 'yes' and false for 'no' and anything else will loop again, except a 'Ctrl C' which stops the whole execution of the program

Reply:
:+1:  I forgot that there is not `while` on Go :sweat_smile: